populator-machinery/controller_test.go

```diff
@@ -593,111 +739,177 @@ func TestSyncPvcWithPopulatorPod(t *testing.T) {
+			expectedKeys:   []string{podKey, pvcPrimeKey, pvKey},
+			expectedObjects: &vpObjects{
+				pvc: pvc(testPvcName, testPvcNamespace, testNodeName, testStorageClassName, "", testPvcUid, []string{pvFinalizer, vpFinalizer},
+					dsf(testApiGroup, testDatasourceKind, testDataSourceName, testPvcNamespace), "", v1.ReadWriteOnce),
+				pvcPrime: pvc(testPvcPrimeName, testVpWorkingNamespace, testNodeName, testStorageClassName, testPvName, testPvcUid, []string{pvFinalizer}, nil, "", v1.ReadWriteOnce),
+				pod:      pod(corev1.PodSucceeded, testNodeName),
 			},
-			populatorArgs:   populatorArgs,
-			expectedResult:  nil,
-			expectedKeys:    []string{podKey, pvcPrimeKey, pvKey},
-			expectedObjects: vpObjects{pvcPrime: pvc(testPvcPrimeName, testVpWorkingNamespace, "", testStorageClassName, testPvName, nil, "", v1.ReadWriteOnce)},
 		},
 		{
 			name:         "Wait for the bind controller to rebind the PV",
 			key:          "pvc/" + testPvcNamespace + "/" + testPvcName,
 			pvcNamespace: testPvcNamespace,
 			pvcName:      testPvcName,
 			initialObjects: []runtime.Object{
-				pvc(testPvcName, testPvcNamespace, testNodeName, testStorageClassName, "",
+				pvc(testPvcName, testPvcNamespace, testNodeName, testStorageClassName, "", testPvcUid, []string{pvFinalizer, vpFinalizer},
 					dsf(testApiGroup, testDatasourceKind, testDataSourceName, testPvcNamespace), "", v1.ReadWriteOnce),
 				ust(),
 				sc(testStorageClassName, storagev1.VolumeBindingWaitForFirstConsumer),
-				pod(corev1.PodSucceeded),
-				pvc(testPvcPrimeName, testVpWorkingNamespace, "", testStorageClassName, testPvName, nil, "", v1.ReadWriteOnce),
-				pv(testPvcName, testPvcNamespace, testPvcUid),
+				pvc(testPvcPrimeName, testVpWorkingNamespace, testNodeName, testStorageClassName, testPvName, testPvcUid, []string{pvFinalizer}, nil, "", v1.ReadWriteOnce),
+				pv(testPvcPrimeName, testVpWorkingNamespace, testPvcUid),
+				pod(corev1.PodSucceeded, testNodeName),
+			},
+			populatorArgs:  populatorArgs,
+			expectedResult: nil,
+			expectedKeys:   []string{podKey, pvcPrimeKey, pvKey},
+			expectedObjects: &vpObjects{
+				pvc: pvc(testPvcName, testPvcNamespace, testNodeName, testStorageClassName, "", testPvcUid, []string{pvFinalizer, vpFinalizer},
+					dsf(testApiGroup, testDatasourceKind, testDataSourceName, testPvcNamespace), "", v1.ReadWriteOnce),
+				pvcPrime: pvc(testPvcPrimeName, testVpWorkingNamespace, testNodeName, testStorageClassName, testPvName, testPvcUid, []string{pvFinalizer}, nil, "", v1.ReadWriteOnce),
+				pv:       pv(testPvcName, testPvcNamespace, testPvcUid),
+				pod:      pod(corev1.PodSucceeded, testNodeName),
 			},
-			populatorArgs:   populatorArgs,
-			expectedResult:  nil,
-			expectedKeys:    []string{podKey, pvcPrimeKey, pvKey},
-			expectedObjects: vpObjects{pvcPrime: pvc(testPvcPrimeName, testVpWorkingNamespace, "", testStorageClassName, testPvName, nil, "", v1.ReadWriteOnce)},
 		},
 		{
 			name:         "Clean up populator pod and pvcPrime",
```

Comment:
Can you rename the tests that are actually deleting  temp resources as well? For example "Clean up populator pod , pvcPrime and temporary resources"

Reply:
Good call! Updated